### PR TITLE
Updating the setup script to handle sample data import for Docker

### LINKED ISSUE #2270: Setup script doesn't handle Docker intuitively
**Describe the bug**

1. The setup script doesn't
    1. remember whether the user running the script previously answered `Y` to this question:
        ```
        Are you setting up this project using Docker? (y/N)
        ```
    1. doesn't detect whether docker is running the application and if so doesn't prompt the user to import the sample data. This means that the user has to refer to INSTALLATION.md and manually do the importation. This is not a good user experience.

**To Reproduce**

1. Run the setup script, select Docker
1. You never get prompted for data import if docker is running the application
1. The script doesn't store your previous Docker related answers

**Expected behavior**

1. Run the setup script, select Docker
1. Get prompted for data import if docker is running the application
1. The script stores your previous Docker related answers
1. Sample data is correctly imported
1. Update `INSTALLATION.md` to:
    1.  let people know that after starting docker for the first time that they should run the installation script again
    1. show people how to install sample data from the CLI when docker is being used.
1. There must be a docker specific way to manually import data successfully with appropriate errors if it fails.
1. The existing non-docker data importation script must fail with an appropriate error if the non-docker API instance and database is not. It does not do that now. Docker can be running and the non-docker data importation runs without failing or giving appropriate messaging.
1. Tests are written to ensure the script always has the newly desired behavior

**Actual behavior**

- See above

**Screenshots**

image

**Additional details**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Suyash878**
**What kind of change does this PR introduce?**

Bug Fix

**Issue Number:**

Fixes #2270 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**

Not sure

**Summary**

The setup script will now prompt the user when it starts the process for sample data import provided the user is using Docker.

**Does this PR introduce a breaking change?**

No

**Other information**

This PR was supposed to be made against the `develop-postgres` branch but since it is currently under work, and unavailability of the setup script there I am currently raising the PR against the `develop` branch.
My changes may get introduced to the `develop-postgres` branch as a seperate issue later on.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Introduced a function for managing Docker container execution with real-time logging.
	- Added prompts for starting Docker containers and importing sample data during setup.
	- New environment variable `DOCKER` added for enhanced configuration options.

- **Bug Fixes**
	- Improved error handling for Docker container startup and sample data import processes.
	- Enhanced logic to ensure MongoDB is ready before proceeding with data imports.

### CODE REVIEW

**Comment by varshith257** [comment]
@Suyash878 Isn't sample data is imported in docker mango container?

**Comment by Suyash878** [comment]
Yes, sample data would be imported to the docker mongo container, totally depends on the mongodb url in the environment variables.

**Comment by Suyash878** [comment]
The base branch check is failing since it is not targeted to the postgres branch.
But I cannot since postres does not have a setup script yet, And the other "sensitive file have been changed" is failing since I made a change to the setup script.

**Comment by Suyash878** [comment]
Sure I will look into the changes.

**Comment by Suyash878** [comment]
I am currently writing tests for these changes.

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by Suyash878** [comment]
The testing Application test is failing due to a bug I think, I've come across this a bunch of times and its something that needs fixing.

**Comment by palisadoes** [comment]
The mongodb connection timed out. I'm rerunning the tests.

**Comment by palisadoes** [comment]
1. It appears none of the past recent PRs that were merged into this branch failed. Please verify this to see whether we merged any PRs with failing tests.
2. If not, you'll need to review your changes to setup.ts to see how it may be impacting the outcome.

**Comment by Suyash878** [comment]
I will look into it.

File: setup.ts [hunk 1]
```diff
@@ -462,6 +462,50 @@
...
+        process.platform === 'win32' ? 'docker.exe' : 'docker',
+        ['info'],
+        { stdio: 'ignore' }
+      );
+      dockerCheck.on('error', reject);
+      dockerCheck.on('close', code => code === 0 ? resolve(null) : reject(new Error('Docker daemon not running')));
+    });
+  } catch (error) {
+    throw new Error('Docker daemon is not running. Please start Docker and try again.'+ error);
+  }
```
[the remark this answers is not in the document]
Comment (Suyash878) [reply setup.ts]: For this particular I will need to define type for error, would it be correct if I assign it as "any", which would make the code line to look something like this:
line 502: `catch(error:any)`

**Comment by Suyash878** [comment]
https://github.com/PalisadoesFoundation/talawa-api/actions/runs/12165995595/job/33931311975?pr=2691
Same error here:
image

**Comment by Suyash878** [comment]
I made another [PR](https://github.com/PalisadoesFoundation/talawa-api/pull/2743) just today, and the "Testing Application" test is failing there as well, and they are both failing due to similar reasons.
You can refer to the failed test for that issue [here](https://github.com/PalisadoesFoundation/talawa-api/actions/runs/12194595393/job/34018770423?pr=2743):

**Comment by palisadoes** [comment]
@Suyash878 Please fix the test so that this PR passes. 

This PR was supposed to have fixed the issue:

- https://github.com/PalisadoesFoundation/talawa-api/commit/37cfbde6667e67e98b233ab65bf4dc2ab22052a9

**Comment by palisadoes** [comment]
@VanshikaSabharwal You have been working on Docker in Talawa Admin

- https://github.com/PalisadoesFoundation/talawa-admin/pull/2615

Can you provide any immediate assistance? This has reached a level of urgency. If so please open a PR against this issue as an emergency override of our policy.

- https://github.com/PalisadoesFoundation/talawa-api/issues/2270

**Comment by VanshikaSabharwal** [comment]
Yes @palisadoes i am doing it

**Comment by palisadoes** [comment]
Please make code rabbit approve your changes

**Comment by VanshikaSabharwal** [comment]
@Suyash878  To approve code by code rabbit you have to make necessary changes recommended by code rabbit in you codebase and resolve each conversation

**Comment by VanshikaSabharwal** [comment]
@Suyash878  I went through the changes you made but i can't see a way to import data for docker using setup script. Can you explain the workflow how you are tackling this issue. Have you made any changes in Docker-compose.dev.yaml

**Comment by Suyash878** [comment]
@palisadoes Shall I create a separate issue for the tests regarding these changes? Considering it would be a cleaner approach that way and since this discussion thread has already been quite cluttered with the main changes themselves.

**Comment by palisadoes** [comment]
We cannot merge into the code base if there are errors. I'll close this until you figure out the solution. However we need this fixed urgently, so I'll open the creation of PRs for this issue to others. @VanshikaSabharwal had volunteered to fix it for example.

**Pull request closed without merging**